src/nginx/transcoded_grpc_server_call.cc

```diff
@@ -208,15 +210,20 @@ bool NgxEspTranscodedGrpcServerCall::ConvertResponseMessage(
   return ReadTranslatedResponse(out);
 }
 
-bool NgxEspTranscodedGrpcServerCall::ReadTranslatedResponse(ngx_chain_t *out) {
+bool NgxEspTranscodedGrpcServerCall::ReadTranslatedResponse(ngx_chain_t **out) {
   // Allocate an ngx_buf.
   ngx_buf_t *buf = reinterpret_cast<ngx_buf_t *>(ngx_calloc_buf(r_->pool));
-  if (!buf) {
+  ngx_chain_t *cl = reinterpret_cast<ngx_chain_t *>(
+      ngx_palloc(r_->pool, sizeof(ngx_chain_t)));
+  if (!buf || !cl) {
```

Comment:
What happens if only one of them fails to allocate? 
Will there be any memory issue? The request will finalize anyway and everything form the pool will be deallocated?

Reply:
the request will fail from here since it return nullptr.  Since all allocated memory are from the request pool, they will be freed.